### PR TITLE
Create microsoft-analysis-tool.js

### PR DESCRIPTION
**Pull request by timmy-ms**
Adding support for analysis tools for MSVC compilers. This will give developers the ability to run with /analyze: when using the Microsoft compilers.

### CODE REVIEW

File: lib/tooling/microsoft-analysis-tool.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2018, Compiler Explorer Authors
```
Comment (RubenRBS) [inline lib/tooling/microsoft-analysis-tool.js:1]: Header needs updating

**Review by RubenRBS — APPROVED** [review]

**Comment by partouf** [comment]
Also needs adding to https://github.com/compiler-explorer/compiler-explorer/blob/main/lib/tooling/_all.js#L37

**Comment by timmy-ms** [comment]
> Also needs adding to https://github.com/compiler-explorer/compiler-explorer/blob/main/lib/tooling/_all.js#L37

Added

**Comment by timmy-ms** [comment]
Doh, wrong button. Reopening, sorry for the noise.

**Review by RubenRBS — APPROVED** [review]

**Comment by partouf** [comment]
@timmy-ms are you using a tool.exe that just points to 1 version now and is that ok?

File: lib/tooling/microsoft-analysis-tool.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        const includeflags = super.getIncludeArguments(compilationInfo.libraries, supportedLibraries);
+
+        let compileFlags = utils.splitArguments(compilationInfo.compiler.options);
+        compileFlags = compileFlags.concat(includeflags);
+        compileFlags = compileFlags.concat(libOptions);
+
+        const manualCompileFlags = options.filter(option => (option !== sourcefile));
+        compileFlags = compileFlags.concat(manualCompileFlags);
+        compileFlags = compileFlags.concat("/analyze:plugin EspXEngine.dll");
+        compileFlags = compileFlags.concat(this.tool.options);
```
Comment (partouf) [inline lib/tooling/microsoft-analysis-tool.js:55]: @timmy-ms compileFlags are not used by super.runTool, are they obsolete or is something wrong with the runTool arguments?

Comment (d-winsor) [reply lib/tooling/microsoft-analysis-tool.js:55]: everything here should be pass to the compiler here as we are just calling MSVC for analysis, not an external tool. compileFlags can be removed and replaced with a concat to args.

**Pull request merged**